include/dali/core/tensor_view.h

```diff
@@ -328,6 +328,23 @@ struct TensorListViewBase {
     calculate_pointers(this->data, data, shape);
   }
 
+  bool is_contiguous() const {
+    if (num_samples() < 2)
+      return true;
+    auto *ptr = data[0];
+    for (int i = 1; i < num_samples(); i++) {
+      auto *next = ptr + volume(tensor_shape_span(i - 1));
+      if (data[i] != next)
+        return false;
+      ptr = next;
+    }
+    return true;
+  }
+
+  bool is_tensor() const {
+    return is_uniform(shape) && is_contiguous();
+  }
```

Comment:
This confuses me a little bit. Why should we call a TensorList, that happens to have every tensor with the same dimensionality and contiguous memory a Tensor? I'd say this still is a list (or vector in that case) of tensors. Typically, TensorList represents a batch in our case, this way should batch be represented by a single Tensor?

Reply:
`CouldBeRepresentedAsASingleTensor` is not a good name, is it? I can put a comment, though, explaining what this function does - or remove it, maybe, because I hanen't used it after all.